semver/is_semver.ts

```diff
@@ -25,22 +25,28 @@ export function isSemVer(value: unknown): value is SemVer {
   if (value === INVALID) return true;
   if (value === ANY) return true;
 
-  const { major, minor, patch, build = [], prerelease = [] } = value as Record<
-    string,
-    unknown
-  >;
-  const result = typeof major === "number" && isValidNumber(major) &&
+  const {
+    major,
+    minor,
+    patch,
+    build = [],
+    prerelease = [],
+  } = value as Record<string, unknown>;
+  return (
+    typeof major === "number" && isValidNumber(major) &&
     typeof minor === "number" && isValidNumber(minor) &&
     typeof patch === "number" && isValidNumber(patch) &&
```

Comment:
Do we need `typeof part === "number"` if we're calling `isValidNumber(part)` anyway?

Reply:
You are right, that is not necessary.